persist/sqldb/workflow_archive.go

```diff
@@ -99,9 +99,19 @@ func (r *workflowArchive) ArchiveWorkflow(wf *wfv1.Workflow) error {
 	logCtx := log.WithFields(log.Fields{"uid": wf.UID, "labels": wf.GetLabels()})
 	logCtx.Debug("Archiving workflow")
 	wf.ObjectMeta.Labels[common.LabelKeyWorkflowArchivingStatus] = "Persisted"
-	workflow, err := json.Marshal(wf)
-	if err != nil {
-		return err
+	var workflow []byte
+	var err error
+	if r.dbType == Postgres {
+		workflow, err = jsonMarshallRawStrings(wf)
```

Comment:
would be good to ensure the new marshalled json is the same as the old way (with exception of the null char), could there be some if condition that compares them? otherwise there is risk that null character gets solved but other characters start breaking

Reply:
My PR has a test for this: https://github.com/argoproj/argo-workflows/pull/14011

Reply:
@MasonM that is only for 1 tiny workflow, in the wild there are all kinds of workflows with various funky characters. i think a runtime check to ensure correctness is better than risk of a different/incorrect workflow being archived/retried

Reply:
@tooptoop4 @MasonM sounds like maybe its better to serialize first and then remove null bytes instead of working at this layer? I was hoping to avoid that but perhaps that is indeed better.